src/services/creategeojsonbloodhound.js

```diff
@@ -58,7 +61,14 @@ ngeo.createGeoJSONBloodhound = function(options, opt_featureProjection,
       },
       filter: function(parsedResponse) {
         var featureCollection = /** @type {GeoJSONFeatureCollection} */
-            (parsedResponse);
+            (goog.object.clone(parsedResponse));
+
+        if (goog.isDef(opt_filter) && !goog.isNull(opt_filter)) {
+          var features = /* Array.<GeoJSONFeature> */
+              (featureCollection['features']);
```

Comment:
We usually rely on the `goog.array` functions. So you may want to use `goog.array.filter` here. And we could use `goog.array.ASSUME_NATIVE_FUNCTIONS=true` in the build config files, to make the compiler actually use the native function in the compiled code.